stdlib/imaplib.pyi

```diff
@@ -151,10 +151,10 @@ class IMAP4_stream(IMAP4):
         def open(self, host: str | None = ..., port: int | None = ...) -> None: ...
 
 class _Authenticator:
-    mech: Callable[[bytes], bytes]
-    def __init__(self, mechinst: Callable[[bytes], bytes]) -> None: ...
+    mech: Callable[[bytes], bytes | bytearray | None]
+    def __init__(self, mechinst: Callable[[bytes], bytes | bytearray | None]) -> None: ...
     def process(self, data: str) -> str: ...
-    def encode(self, inp: bytes) -> str: ...
+    def encode(self, inp: bytes | bytearray) -> str: ...
```

Comment:
I think this is `str | (Buffer & Sequence)`, e.g. don't see anything stopping `memoryview` from working here